typespec-tests/tsp/multipart.tsp

```diff
@@ -35,3 +35,25 @@ op upload(
   @header contentType: "multipart/form-data",
   @body data: FormData,
 ): OkResponse;
+
+model Inherit1File extends File {
+  contentType?: string;
+  filename: string;
+}
+
+model Inherit2File extends Inherit1File {
+  contentType: "image/jpeg";
+}
+
+@route("/uploadHttpPart/images/{name}")
+@post
+op uploadHttpPart(
+  @path name: string,
+  @query compress?: boolean,
+  @header contentType: "multipart/form-data",
+  @multipartBody body: {
+    fileData1: HttpPart<Inherit2File>;
+    fileData2: HttpPart<File>;
```

Comment:
2 test cases.

1st for inherit and override of default property in `File`.
2nd for just using the `File` from http lib.